diff --git a/dist/memory.cjs b/dist/memory.cjs
index f5bceead9a5533ff71e0c6f6960384ced2ee0060..5085269a58493c29e36ea7384368cd64a0e979a5 100644
--- a/dist/memory.cjs
+++ b/dist/memory.cjs
@@ -41,7 +41,7 @@ exports.getInputValue = getInputValue;
  */
 const getOutputValue = (outputValues, outputKey) => {
     const value = getValue(outputValues, outputKey);
-    if (!value) {
+    if (!value && value !== '') {
         const keys = Object.keys(outputValues);
         throw new Error(`output values have ${keys.length} keys, you must specify an output key or pass only 1 key as output`);
     }
diff --git a/dist/memory.js b/dist/memory.js
index 8e221f55b151d0ff7592b30ca34363b9ec577e01..a3ff2a514eda6c7e606e8f6d7305535b29ab671e 100644
--- a/dist/memory.js
+++ b/dist/memory.js
@@ -36,7 +36,7 @@ export const getInputValue = (inputValues, inputKey) => {
  */
 export const getOutputValue = (outputValues, outputKey) => {
     const value = getValue(outputValues, outputKey);
-    if (!value) {
+    if (!value && value !== '') {
         const keys = Object.keys(outputValues);
         throw new Error(`output values have ${keys.length} keys, you must specify an output key or pass only 1 key as output`);
     }
