diff --git a/js/helper.js b/js/helper.js
index 37798b9444d39a8713327ed12adf3e76f03188a4..08d49b7a5c058c4d29c9929c0c524c85b14b330e 100644
--- a/js/helper.js
+++ b/js/helper.js
@@ -15,6 +15,7 @@ function getBinaryPath() {
   const parts = [];
   parts.push(__dirname);
   parts.push('..');
+  parts.push('bin');
   parts.push(`sentry-cli${process.platform === 'win32' ? '.exe' : ''}`);
   return path.resolve(...parts);
 }