diff --git a/index.d.ts b/index.d.ts
index 7a8182a94289524851cb08a3b24897f2b6bce747..f5bfb61bdacbae81ca274cc4b5a61e6e7322b7cd 100755
--- a/index.d.ts
+++ b/index.d.ts
@@ -72,6 +72,9 @@ declare class WebSocket extends EventEmitter {
         | typeof WebSocket.CLOSED;
     readonly url: string;
 
+    /** Indicates if the connection has replied to the last PING */
+    isAlive: boolean;
+
     /** The connection is not yet open. */
     readonly CONNECTING: 0;
     /** The connection is open and ready to communicate. */