diff --git a/dist/rules/consistent-type-imports.js b/dist/rules/consistent-type-imports.js
index 1844dc32b19d10abbe13556b9ee2f69c0aabac05..01320c5212fd61e08ca8a438db3ccd59949f9421 100644
--- a/dist/rules/consistent-type-imports.js
+++ b/dist/rules/consistent-type-imports.js
@@ -81,6 +81,8 @@ exports.default = util.createRule({
                 ImportDeclaration(node) {
                     var _a;
                     const source = node.source.value;
+                    if (source.endsWith('.vue')) return;
+
                     // sourceImports is the object containing all the specifics for a particular import source, type or value
                     const sourceImports = (_a = sourceImportsMap[source]) !== null && _a !== void 0 ? _a : (sourceImportsMap[source] = {
                         source,