diff --git a/dist/axios.js b/dist/axios.js
index f1142b7..d254086 100644
--- a/dist/axios.js
+++ b/dist/axios.js
@@ -4692,7 +4692,7 @@
 
                   // slice off the Error: ... line
                   stack = function () {
-                    if (!dummy.stack) {
+                    if (typeof dummy.stack !== 'string') {
                       return '';
                     }
                     var firstNewlineIndex = dummy.stack.indexOf('\n');
diff --git a/dist/browser/axios.cjs b/dist/browser/axios.cjs
index 916b539..219403c 100644
--- a/dist/browser/axios.cjs
+++ b/dist/browser/axios.cjs
@@ -4558,7 +4558,7 @@ class Axios {
 
         // slice off the Error: ... line
         const stack = (() => {
-          if (!dummy.stack) {
+          if (typeof dummy.stack !== 'string') {
             return '';
           }
 
diff --git a/dist/esm/axios.js b/dist/esm/axios.js
index 19dc35f..89a4559 100644
--- a/dist/esm/axios.js
+++ b/dist/esm/axios.js
@@ -4556,7 +4556,7 @@ let Axios$1 = class Axios {
 
         // slice off the Error: ... line
         const stack = (() => {
-          if (!dummy.stack) {
+          if (typeof dummy.stack !== 'string') {
             return '';
           }
 
diff --git a/dist/node/axios.cjs b/dist/node/axios.cjs
index 46078b3..9ef4bbd 100644
--- a/dist/node/axios.cjs
+++ b/dist/node/axios.cjs
@@ -5428,7 +5428,7 @@ class Axios {
 
         // slice off the Error: ... line
         const stack = (() => {
-          if (!dummy.stack) {
+          if (typeof dummy.stack !== 'string') {
             return '';
           }
           const firstNewlineIndex = dummy.stack.indexOf('\n');
diff --git a/lib/core/Axios.js b/lib/core/Axios.js
index 7fc529e..97705a5 100644
--- a/lib/core/Axios.js
+++ b/lib/core/Axios.js
@@ -47,7 +47,7 @@ class Axios {
 
         // slice off the Error: ... line
         const stack = (() => {
-          if (!dummy.stack) {
+          if (typeof dummy.stack !== 'string') {
             return '';
           }
 
