diff --git a/dist/main.js b/dist/main.js
index 86be84f44210b26583e0a7f1732acd8b98a5e701..a2b05be6a45355704fedf43b51a34793580eaf6c 100644
--- a/dist/main.js
+++ b/dist/main.js
@@ -42,7 +42,7 @@ $parcel$export(module.exports, "verifyChallenge", () => $f5bfd4ce37214f4f$export
  * @param size The desired length of the string
  * @returns The random string
  */ function $f5bfd4ce37214f4f$var$random(size) {
-    const mask = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._~";
+    const mask = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._";
     let result = "";
     const randomUints = $f5bfd4ce37214f4f$var$getRandomValues(size);
     for(let i = 0; i < size; i++){