diff --git a/cjs/container-instance.class.js b/cjs/container-instance.class.js
index e473b1e652aa0b6e7462f7ba93fcef2812483b20..411dd978ca5b29d5285bdd9359624761ea8b5c70 100644
--- a/cjs/container-instance.class.js
+++ b/cjs/container-instance.class.js
@@ -234,6 +234,9 @@ class ContainerInstance {
      */
     initializeParams(target, paramTypes) {
         return paramTypes.map((paramType, index) => {
+            if (paramType === undefined) {
+                throw new ReferenceError(`Circular dependency: Target: ${target.name}. Index: ${index} `);
+            }
             const paramHandler = container_class_1.Container.handlers.find(handler => {
                 /**
                  * @Inject()-ed values are stored as parameter handlers and they reference their target
