# Enforce correct package naming convention for n8n community nodes (`@n8n/community-nodes/package-name-convention`)

üíº This rule is enabled in the following configs: ‚úÖ `recommended`, ‚òëÔ∏è `recommendedWithoutN8nCloudSupport`.

üí° This rule is manually fixable by [editor suggestions](https://eslint.org/docs/latest/use/core-concepts#rule-suggestions).

<!-- end auto-generated rule header -->

## Rule Details

Validates that your package name follows the correct n8n community node naming convention. Package names must start with `n8n-nodes-` and can optionally be scoped.

## Examples

### ‚ùå Incorrect

```json
{
  "name": "my-service-integration"
}
```

```json
{
  "name": "nodes-my-service"
}
```

```json
{
  "name": "@company/my-service"
}
```

### ‚úÖ Correct

```json
{
  "name": "n8n-nodes-my-service"
}
```

```json
{
  "name": "@company/n8n-nodes-my-service"
}
```

## Best Practices

- Use descriptive service names: `n8n-nodes-github` rather than `n8n-nodes-api`
- For company packages, use your organization scope: `@mycompany/n8n-nodes-internal-tool`
